src/rspamadm/configdump.c

```diff
@@ -305,6 +308,119 @@ rspamadm_configdump (gint argc, gchar **argv, const struct rspamadm_command *cmd
 			exit (EXIT_SUCCESS);
 		}
 
+		if (symbol_full_details) {
+			/*
+			 * Create object from symbols groups and output it using the
+			 * specified format
+			 */
+			ucl_object_t *out = ucl_object_typed_new (UCL_OBJECT);
+			GHashTableIter it;
+			gpointer sk, sv;
+
+			g_hash_table_iter_init (&it, cfg->symbols);
+			ucl_object_t *sym_ucl = ucl_object_typed_new (UCL_OBJECT);
+			const ucl_object_t *all_symbols_ucl = ucl_object_lookup(cfg->rcl_obj, "symbols");
+
+			while (g_hash_table_iter_next (&it, &sk, &sv)) {
+				const gchar *sym_name = (const gchar *)sk;
+				struct rspamd_symbol *s = (struct rspamd_symbol *)sv;
+				ucl_object_t *this_sym_ucl = ucl_object_typed_new (UCL_OBJECT);
+
+				ucl_object_insert_key (this_sym_ucl,
+						ucl_object_fromdouble (s->score),
+						"score", strlen ("score"),
+						false);
+				ucl_object_insert_key (this_sym_ucl,
+						ucl_object_fromstring (s->description),
+						"description", strlen ("description"), false);
+
+				rspamd_symcache_get_symbol_details(cfg->cache, sym_name, this_sym_ucl);
+
+				ucl_object_insert_key (this_sym_ucl,
+						ucl_object_frombool (!!(s->flags & RSPAMD_SYMBOL_FLAG_DISABLED)),
+						"disabled", strlen ("disabled"),
+						false);
+
+				if (s->nshots == 1) {
+					ucl_object_insert_key (this_sym_ucl,
+							ucl_object_frombool (true),
+							"one_shot", strlen ("one_shot"),
+							false);
+				}
+				else {
+					ucl_object_insert_key (this_sym_ucl,
+							ucl_object_frombool (false),
+							"one_shot", strlen ("one_shot"),
+							false);
+				}
+
+				if (s->gr != NULL) {
+					struct rspamd_symbols_group *gr = s->gr;
+					const char *gr_name = gr->name;
+					if (strcmp(gr_name, "ungrouped") != 0) {
+						ucl_object_insert_key (this_sym_ucl,
+							ucl_object_fromstring (gr_name),
+							"group", strlen("group"),
+							false);
+					}
+
+					if (s->groups) {
+						ucl_object_t *add_groups = ucl_object_typed_new (UCL_ARRAY);
+						guint j;
+						struct rspamd_symbols_group *add_gr;
+						bool one_extra = false;
```

Comment:
I would suggest to rename `one_extra` to something like `has_extra_groups` as it is more descriptive I think.

Reply:
Changed.